### PR TITLE
Add containers for building and testing SU2 with the thread sanitizer

### PR DESCRIPTION
**Pull request by jblueh**
## Overview

This PR provides a container that can be used for building SU2 with the thread sanitizer. Further steps would be a container for testing with the thread sanitizer, and ultimately running thread sanitizer tests as a part of SU2's CI pipeline. This is work in progress and there are things that need testing, hence the focus on the build container for now.

1) Regarding the workflow in this repo that builds and uploads container images, do the resources of the CI pipeline suffice? The container involves building `gcc` from source, which does take some time.
2) The existing build container and the new build container have common parts, when we know that everything works we could consider unifying them.

### CODE REVIEW

File: build/Dockerfile.tsan [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Ensure that ccache uses the built gcc
+RUN export PATH=/gcc_install/bin:$PATH \
+ && /usr/sbin/update-ccache-symlinks
+
+# Configure environment for the built gcc, also enable thread sanitizer
+ENV LD_LIBRARY_PATH=/gcc_install/lib64:$LD_LIBRARY_PATH
+ENV CXXFLAGS="-I/gcc_install/include/c++/9.4.0 -I/gcc_install/include/c++/9.4.0/x86_64-pc-linux-gnu -fsanitize=thread"
+ENV CFLAGS="-I/gcc_install/include/c++/9.4.0 -I/gcc_install/include/c++/9.4.0/x86_64-pc-linux-gnu -fsanitize=thread"
+ENV LDFLAGS="-L/gcc_install/lib64 -fsanitize=thread"
+ENV PATH=/usr/lib/ccache:/gcc_install/bin:$PATH
```
Comment (jblueh) [inline build/Dockerfile.tsan:58]: In the non-tsan build container, `ccache` is added to the path via `bashrc`. Was there a specific reason for this?

Comment (pcarruscag) [reply build/Dockerfile.tsan:58]: Not that I know, but I don't know much about the containers :)
Who does git blame for that line?

**Review by pcarruscag — APPROVED** [review]
Do you want to add this to the actions https://github.com/su2code/Docker-Builds/blob/master/.github/workflows/docker-image-upload.yml
So that we know whether it builds and uploads fine? Then it would be easy to use in our tests.

File: build/Dockerfile.tsan (hunk 1 above)
Comment (jblueh) [reply build/Dockerfile.tsan:58]: That would be @talbring

**Comment by jblueh** [comment]
> Do you want to add this to the actions https://github.com/su2code/Docker-Builds/blob/master/.github/workflows/docker-image-upload.yml So that we know whether it builds and uploads fine? Then it would be easy to use in our tests.

Yes, I will add it.

File: build/Dockerfile.tsan (hunk 1 above)
Comment (pcarruscag) [reply build/Dockerfile.tsan:58]: He may not remember any more, it has been a while :) we can always change it and test if you want.

Comment (jblueh) [reply build/Dockerfile.tsan:58]: Might be something to revisit if we unify the build containers. :+1:

**Comment by jblueh** [comment]
Should I go ahead and merge to trigger the pipeline?

**Comment by pcarruscag** [comment]
Yeah go for it

**Pull request merged**